pkg/hive/health/metrics.go

```diff
@@ -71,10 +91,19 @@ func publishJob(ctx context.Context, p metricPublisherParams, publish publishFun
 	it, watch := p.Table.AllWatch(p.DB.ReadTxn())
 	for {
 		stats := make(map[types.Level]uint64)
+		idToStatus := make(map[string]uint64)
 		for obj := range it {
 			stats[obj.Level]++
+
+			_, ok := idToStatus[obj.ID.Module.String()]
+			if obj.Level == types.LevelDegraded {
+				idToStatus[obj.ID.Module.String()]++
+			} else if !ok {
+				idToStatus[obj.ID.Module.String()] = 0
```

Comment:
How does this work with duplicate module ids? Say we have two modules ”a.c.foo.reconciler” and ”a.c.bar.reconciler” and one is healthy and one is degraded. How would that show up?

Reply:
`ID.Module` is actually the  `FullModuleID is the fully qualified module identifier, e.g. the concat of nested module ids, e.g. "agent.controlplane.endpoint-manager". Provided in the module's scope.`

So we should mark as degraded, `a.c.foo` or `a.c.bar`. We are marking as degraded module not the components if that makes sense?

Take the following example:

`ID.Module.String()`: `agent.controlplane.cilium-agent-dynamic-config`
`ID.String()`: `agent.controlplane.cilium-agent-dynamic-config.job-k8s-reflector-cilium-configs-cm-cilium-config-kube-system`